master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -261,8 +261,20 @@ private[celeborn] class Master(
       }).sum()
   }
 
+  masterSource.addGauge(MasterSource.SHUFFLE_TOTAL_COUNT) { () =>
+    statusSystem.shuffleTotalCount.sum()
+  }
+
   masterSource.addGauge(MasterSource.SHUFFLE_FALLBACK_COUNT) { () =>
-    statusSystem.shuffleTotalFallbackCount.sum()
+    statusSystem.shuffleFallbackCounts.values().asScala.map(_.longValue()).sum
+  }
+
+  statusSystem.shuffleFallbackCounts.entrySet().asScala.foreach { shuffleFallbackCount =>
```

Comment:
If the initial `shuffleFallbackCounts` is empty, it will not register the gauge metrics

Reply:
how to register metrics during runtime?

Reply:
seems we can register metrics on demand when add new key into the map

Reply:
@turboFei, thanks for reminder. I have updated the registration after `statusSystem.handleAppHeartbeat`.